### ISSUE TITLE
VS 2013 compilation error (C1045: compiler limitations: the link specification is too nested)

### ISSUE THREAD

**Issue by cjmxp** [godotengine/godot#10219]
C:\MyWork\godot\modules\gdnative\godot/vector3.h (34): fatal, error, C1045: compiler
Limitations: the link specification is too nested

**Comment by akien-mga** [comment]
It's a bit hard to say anything with so little output.

**Comment by cjmxp** [comment]
这是所有的错误输出
This is the output of all errors

**Comment by cjmxp** [comment]
This is the output of all errors

**Comment by Zireael07** [comment]
@cjmxp: What are you using to build? Scons or mingw? If scons, are you using -j (multiple cores)?

**Comment by cjmxp** [comment]
@Zireael07 vs2013 tools command  
scons p=windows

**Comment by rubenjavier** [comment]
Can confirm this too, Windows 10 64bit, commit 5052cb2b9104f57d0ff38d80da50bbeceb926e6f

my compilation command (I've tried with and without -j param and doing a clean before compiling)
```
"C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\vcvarsall.bat" amd64 && C:\Python27\Scripts\scons platform=windows -j 8 tools=yes target=release_debug CCFLAGS=/EHsc
```
the error:
```
modules\gdnative\godot/vector3.h (34): fatal, error, C1045: compiler limitations: the link specification is too nested
```

**Comment by starry-abyss** [comment]
same here, win7 64bit, vs2013 sp5, commit ef08228

[error.txt](https://github.com/godotengine/godot/files/1356360/error.txt)

**Comment by sbeca** [comment]
I'm getting this error as well so I'll add my details to the list in the hope that it will help others track down this issue.

Here is my setup:
- Windows 10 Enterprise 64-bit (latest updates installed)
- Visual Studio Ultimate 2013 - Version 12.0.30723.00 Update 3
- Python 2.7.14 32-bit
- SCons 3.0.0
- Godot 3.0 f52ab8d86418a67ddee247ed7765e72935b0c57f


Here is my `cmd` output:
```
Microsoft Windows [Version 10.0.15063]
(c) 2017 Microsoft Corporation. All rights reserved.

C:\Users\Scott\Documents\GitHub\godot>"C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\vcvarsall.bat" amd64 && scons platform=windows tools=yes target=release_debug
scons: Reading SConscript files ...
Detected MSVC compiler: amd64
Compiled program architecture will be a 64 bit executable (forcing bits=64).
cdir is: C:\Users\Scott\Documents\GitHub\godot
Checking for C header file mntent.h... (cached) no
scons: done reading SConscript files.
scons: Building targets ...
[Initial build] Compiling ==> modules\gdnative\gdnative\dictionary.cpp
dictionary.cpp
modules\gdnative\include\gdnative/vector3.h(34) : fatal error C1045: compiler limit : linkage specifications nested too deeply
scons: *** [modules\gdnative\gdnative\dictionary.windows.opt.tools.64.obj] Error 2
scons: building terminated because of errors.
```

**Comment by akien-mga** [comment]
The easy fix would be to drop support for VS 2013... but before doing that, maybe someone finds a way to make VS 2013 happy without having to drop support for it.

**Comment by starry-abyss** [comment]
Please try my PR ^

**Comment by sbeca** [comment]
Just a quick follow up to say that since 3ba5ee42c3b019234fa154c66683d27e626a164e has been merged, I am now able to build Godot successfully on the computer that was previously having this issue. Thanks!

**Comment by akien-mga** [comment]
Indeed, it's fixed by #12236.
